src/sql/run.py

```diff
@@ -274,19 +273,38 @@ def interpret_rowcount(rowcount):
         result = '%d rows affected.' % rowcount
     return result
 
+class FakeResultProxy(object):
+    """A fake class that pretends to behave like the ResultProxy from
+    SqlAlchemy.
+    """
+    def __init__(self, cursor, headers):
+        self.fetchall = cursor.fetchall
+        self.fetchmany = cursor.fetchmany
+        self.rowcount = cursor.rowcount
+        self.keys = lambda: headers
+        self.returns_rows = True
+
 
 def run(conn, sql, config, user_namespace):
     if sql.strip():
         for statement in sqlparse.split(sql):
-            if sql.strip().split()[0].lower() == 'begin':
+            first_word = sql.strip().split()[0].lower()
+            if first_word == 'begin':
                 raise Exception("ipython_sql does not support transactions")
-            txt = sqlalchemy.sql.text(statement)
-            result = conn.session.execute(txt, user_namespace)
+            if first_word.startswith('\\') and 'postgres' in str(conn.dialect):
+                pgspecial = PGSpecial()
+                _, cur, headers, _ = pgspecial.execute(
+                                              conn.session.connection.cursor(),
```

Comment:
pgspecial needs the raw cursor object psycopg2 in order to use the mogrify() method. So I had to reach into the sqlalchemy connection object to get the raw cursor.